File: Python_Solutions/CodeWars/6kyu/Count_characters_in_your_string.py
# The main idea is to count all the occurring characters in a string. If you have a string like aba,
# then the result should be {'a': 2, 'b': 1}.
#
# What if the string is empty? Then the result should be empty object literal, {}.
#
# STRINGSFUNDAMENTALS
# Solution
def count(string):
    dict = {}
    count = 0
    for char in string:
        count = string.count(char)
        dict[f"{char}"] = count
    return dict

File: CodeWars/6kyu/Count_characters_in_your_string.cpp
/*
The main idea is to count all the occurring characters in a string. If you have a string like aba, then the result should be {'a': 2, 'b': 1}.

What if the string is empty? Then the result should be empty object literal, {}.

STRINGSFUNDAMENTALS
*/
// Solution
#include <map>
#include <string>

std::map<char, unsigned> count(const std::string& string) {
    std::map<char, unsigned> storage;
    for (char letter : string) {
      storage[letter]++;
    }
    return storage;
}